referenceqvm/unitary_generator.py

```diff
@@ -187,16 +190,17 @@ def permutation_arbitrary(args, num_qubits):
         start_i - starting index to lift gate from
     :rtype:  tuple (sparse_array, np.array, int)
     """
-    # Check input
-    if type(args) is tuple or type(args) is list:
-        if len(args) is 0:
-            raise TypeError("Need at least one qubit index to perform"
-                            "permutation")
-    else:
+    # Don't permit NoneType or empty sequences, but allow 0
+    if args is None or (isinstance(args, Sequence) and not args):
```

Comment:
Thanks for the improvements here. `args` is a required argument and hence cannot be none, so we should be able to drop this check?

Reply:
If you set `args` to `None` in `master`, you wind up with `args` being set to `[None]`. I wanted to prevent that situation to the best of my ability. I can remove it, if that situation is acceptable though.